src/main/java/seedu/address/model/item/Item.java

```diff
@@ -0,0 +1,98 @@
+package seedu.address.model.item;
+
+import java.util.Objects;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+/**
+ * Represents an Item in FoodREM.
+ * TODO: Implement Item with Tags
+ */
+public class Item {
+    public final ItemName name;
+    public final ItemQuantity quantity;
+    public final ItemDate boughtDate;
+    public final ItemDate expiryDate;
+
+    /**
+     * Constructs an item.
+     * @param name Name of the item.
+     * @param quantity Quantity of the item.
+     * @param boughtDate Date when the item was purchased.
+     * @param expiryDate Date when the item expires.
+     * TODO: Make expiryDate an optional field.
+     */
+    public Item(ItemName name, ItemQuantity quantity, ItemDate boughtDate, ItemDate expiryDate) {
+       requireAllNonNull(name, quantity, boughtDate, expiryDate);
+       this.name = name;
+       this.quantity = quantity;
+       this.boughtDate = boughtDate;
+       this.expiryDate = expiryDate;
+    }
+
+    public ItemName getName() {
+        return name;
+    }
+
+    public ItemQuantity getQuantity() {
+        return quantity;
+    }
+
+    public ItemDate getBoughtDate() {
+        return boughtDate;
+    }
+
+    public ItemDate getExpiryDate() {
+        return expiryDate;
+    }
+
+    /**
+     * Returns true if both items have the same identity and data fields.
+     * @param other Object to compare against.
+     * @return True if both items have the same identity and data fields.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof Item)) {
+            return false;
+        }
+
+        Item otherItem = (Item) other;
+        return otherItem.getName().equals(getName())
+                && otherItem.getQuantity().equals(getQuantity())
```

Comment:
Should this only be name?
 If items are only equal if they have the same quantity and dates, this would mean that we can created "different" items with same name and just different quantity or bought date. 

Now that I think of it. Is there a way for our application to store same products but bought on different dates?

Reply:
Good point. I think it's fine for now, but I'll create an issue for this.

Reply:
Meeting on 5/20/22: For now, equality check will done only by name.